src/Orleans.Runtime/Silo/Silo.cs

```diff
@@ -187,7 +187,7 @@ public Silo(ILocalSiloDetails siloDetails, IServiceProvider services)
             // Initialize the message center
             messageCenter = Services.GetRequiredService<MessageCenter>();
             var dispatcher = this.Services.GetRequiredService<Dispatcher>();
-            messageCenter.RerouteHandler = dispatcher.RerouteMessage;
+            messageCenter.RerouteHandler = message => dispatcher.SendMessage(message);
```

Comment:
Why not `dispatcher.ResendMessageImpl(message)` as `RerouteMessage` used to do?

Reply:
This logic could do with a refactoring. RerouteHandler is called from RerouteMessage which is only called from GatewayAcceptor.HandleMessage when the message is not already addressed.

So they will have the same effect but this skips a bit of logic that would be skipped anyway.

I'll change it back since it's not the purpose of this PR